### ISSUE TITLE
TypeError: LoadLibrary() argument 1 must be str, not None

### ISSUE THREAD

**Issue by asaelbarilan** [conda/conda#8836]
Hey all, ive been trying to do :
conda create -n myenv python=3.5.1
or  conda list

'''
**TypeError: LoadLibrary() argument 1 must be str, not None''**
'''
## Expected Behavior:
new env creation..
or 
list of all conda pkgs
 
(when i looked for this bugs it gave answers for specific issues it was caused by nothing that i  understood could help me)
 
[details:
windows 10
conda 4.7.5
Python 3.7.3]

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda\exceptions.py", line 1043, in __call__
        return func(*args, **kwargs)
      File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda\cli\main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda\cli\conda_argparse.py", line 80, in do_call
        module = import_module(relative_mod, __name__.rsplit('.', 1)[0])
      File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\importlib\__init__.py", line 127, in import_module
        return _bootstrap._gcd_import(name[level:], package, level)
      File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
      File "<frozen importlib._bootstrap>", line 983, in _find_and_load
      File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
      File "<frozen importlib._bootstrap_external>", line 728, in exec_module
      File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
      File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda\cli\main_info.py", line 19, in <module>
        from ..core.index import _supplement_index_with_system
      File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda\core\index.py", line 9, in <module>
        from .package_cache_data import PackageCacheData
      File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda\core\package_cache_data.py", line 15, in <module>
        from conda_package_handling.api import InvalidArchiveError
      File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda_package_handling\api.py", line 3, in <module>
        from libarchive.exception import ArchiveError as _LibarchiveArchiveError
      File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\libarchive\__init__.py", line 1, in <module>
        from .entry import ArchiveEntry
      File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\libarchive\entry.py", line 6, in <module>
        from . import ffi
      File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\libarchive\ffi.py", line 27, in <module>
        libarchive = ctypes.cdll.LoadLibrary(libarchive_path)
      File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\ctypes\__init__.py", line 434, in LoadLibrary
        return self._dlltype(name)
      File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\ctypes\__init__.py", line 356, in __init__
        self._handle = _dlopen(self._name, mode)
    TypeError: LoadLibrary() argument 1 must be str, not None

`$ C:\Users\asaelb\AppData\Local\Continuum\anaconda3\Scripts\conda-script.py info`


An unexpected error has occurred. Conda has prepared the above report.

Upload successful.
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
C:\Users\asaelb>conda config --show-sources
==> C:\Users\asaelb\.condarc <==
ssl_verify: False
channels:
  - defaults
  - conda-forge
report_errors: True

```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
i get same error for this command too...:

TypeError: LoadLibrary() argument 1 must be str, not None
```
</p></details>

**Comment by jorisvandenbossche** [comment]
I also ran into this on AppVeyor with `conda info -a`:

```
conda info -a
# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<
    Traceback (most recent call last):
      File "C:\Miniconda37-x64\lib\site-packages\conda\exceptions.py", line 1043, in __call__
        return func(*args, **kwargs)
      File "C:\Miniconda37-x64\lib\site-packages\conda\cli\main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "C:\Miniconda37-x64\lib\site-packages\conda\cli\conda_argparse.py", line 80, in do_call
        module = import_module(relative_mod, __name__.rsplit('.', 1)[0])
      File "C:\Miniconda37-x64\lib\importlib\__init__.py", line 127, in import_module
        return _bootstrap._gcd_import(name[level:], package, level)
      File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
      File "<frozen importlib._bootstrap>", line 983, in _find_and_load
      File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
      File "<frozen importlib._bootstrap_external>", line 728, in exec_module
      File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
      File "C:\Miniconda37-x64\lib\site-packages\conda\cli\main_info.py", line 19, in <module>
        from ..core.index import _supplement_index_with_system
      File "C:\Miniconda37-x64\lib\site-packages\conda\core\index.py", line 9, in <module>
        from .package_cache_data import PackageCacheData
      File "C:\Miniconda37-x64\lib\site-packages\conda\core\package_cache_data.py", line 15, in <module>
        from conda_package_handling.api import InvalidArchiveError
      File "C:\Miniconda37-x64\lib\site-packages\conda_package_handling\api.py", line 3, in <module>
        from libarchive.exception import ArchiveError as _LibarchiveArchiveError
      File "C:\Miniconda37-x64\lib\site-packages\libarchive\__init__.py", line 1, in <module>
        from .entry import ArchiveEntry
      File "C:\Miniconda37-x64\lib\site-packages\libarchive\entry.py", line 6, in <module>
        from . import ffi
      File "C:\Miniconda37-x64\lib\site-packages\libarchive\ffi.py", line 27, in <module>
        libarchive = ctypes.cdll.LoadLibrary(libarchive_path)
      File "C:\Miniconda37-x64\lib\ctypes\__init__.py", line 434, in LoadLibrary
        return self._dlltype(name)
      File "C:\Miniconda37-x64\lib\ctypes\__init__.py", line 356, in __init__
        self._handle = _dlopen(self._name, mode)
    TypeError: LoadLibrary() argument 1 must be str, not None
`$ C:\Miniconda37-x64\Scripts\conda-script.py info -a`
```

Full report at https://ci.appveyor.com/project/jorisvandenbossche/geopandas-us6w0/builds/25574390/job/fl6f0buxpexo6nf4

**Comment by msarahan** [comment]
libarchive DLLs are not getting found.  Are you activating the environment with Conda in it?  We had to tweak our appveyor install a bit:

https://github.com/conda/conda/commit/44557fc90c02c95ea3fbd4b95e447c9d44b01303

**Comment by jorisvandenbossche** [comment]
The full script is in https://github.com/geopandas/geopandas/blob/master/appveyor.yml. We currently do a `"set PATH=%MINICONDA%;%MINICONDA%\\Scripts;%PATH%"` before using conda commands.

**Comment by msarahan** [comment]
Yep, that is the problem.  Scripts has conda.exe, not conda.bat, but conda.bat is the thing that sets up the correct library paths.

**Comment by asaelbarilan** [comment]
So did you run the script with miniconda and it works fine?

בתאריך יום ה׳, 27 ביוני 2019, 19:42, מאת Mike Sarahan ‏<
notifications@github.com>:

**Comment by jorisvandenbossche** [comment]
@msarahan just to understand the context a bit better: was this a "wrong" usage in that appveyor script but worked by accident, or is it a regression in conda that needs to be fixed, or was it something that was changed deliberately and users need to update that?

**Comment by msarahan** [comment]
conda 4.6 changed where conda "lives" to improve isolation and to make it possible for conda to not interfere with the other system software.  Anaconda and Miniconda installers do not modify PATH by default.  They instead modify the registry to auto-run a batch script that makes conda (and only conda) available.

The modification of PATH that you are doing there has never been correct.  It has worked by chance.  That's a partial list, and doesn't include things in the correct order.  If you insist on setting PATH, you need to add more entries.  https://github.com/conda/conda/blob/master/conda/activate.py#L493-L519

**Comment by MikhailRyazanov** [comment]
@msarahan, we have a similar problem: `conda info -a` [just crashes](https://ci.appveyor.com/project/DanHickstein/pyabel/builds/25542796/job/9u2csw5kl8b670an#L70) instead of printing
```
     active environment : None
```
as it [did before](https://ci.appveyor.com/project/DanHickstein/pyabel/builds/25490151/job/jo6jqwjp1t5iodc6#L43).

Notice that **4.6**.14 actually did not crash, whereas 4.7.5 crashes. And only in Python 3 — the 4.7.5-**py27**_0 version [still works](https://ci.appveyor.com/project/DanHickstein/pyabel/builds/25542796/job/vxx9ma7hvy013hun#L70).

I'd say that wherever Conda “lives” now, as long as `conda info` is able to start at all, it should always print something useful (at least, a warning) instead of crashing with a cryptic traceback.

**Comment by msarahan** [comment]
PRs would be welcome.  This PATH modification: https://ci.appveyor.com/project/DanHickstein/pyabel/builds/25490151/job/jo6jqwjp1t5iodc6#L8 is completely unsupported.  It has never been an officially supported collection of paths.  It is incomplete.  We're not going to spend time supporting something that was working by chance.  Sorry.

**Comment by MikhailRyazanov** [comment]
So what is the proper way to set it up?

**Comment by msarahan** [comment]
Please see the commit I linked to above.  https://github.com/conda/conda/commit/44557fc90c02c95ea3fbd4b95e447c9d44b01303

**Comment by jorisvandenbossche** [comment]
@msarahan thanks a lot for the explanation!

**Comment by jorisvandenbossche** [comment]
@msarahan the `CALL "%PYTHON_ROOT%\\Scripts\\activate.bat"` solution does not seem to work fully (or I am missing something obvious).

In https://github.com/geopandas/geopandas/pull/1029 I am re-enabling conda 4.7, but in the test_script section of appveyor I now get "*'conda' is not recognized as an internal or external command*" (see eg https://ci.appveyor.com/project/jorisvandenbossche/geopandas-us6w0/builds/25637234/job/1wniitwodpcc0y6h).  
In the 'install' section it worked, and I tried adding a similar CALL in the 'test_script' section, but that doesn't seem to help

**Comment by adriantorrie** [comment]
Try running `conda activate base` before any `conda` commands. Worked for me with exact same `TypeError: LoadLibrary() argument 1 must be str, not None` error

**Comment by leycec** [comment]
If `conda activate base` is now a mandatory prerequisite to running any other `conda` subcommands, shouldn't `conda` itself guarantee this constraint by internally running the equivalent of `conda activate base` when no other environment is currently active? Shouldn't establishing sane defaults already be the default behaviour?

I'm probably missing something here.

Unrelatedly, this changeset significantly broke backward compatibility with many, many Appveyor + `conda` CI pipelines. I acknowledge that manually setting the `%PATH%` was never an officially supported solution. Nonetheless, effectively *every* Appveyor + `conda` CI pipeline configuration I've encountered did so. Given the ubiquity of this issue, it would have been useful if a set of guidelines for transitioning from the old unsupported approach to the new supported approach had been officially published *before* this changeset broke everything for everyone.

In short, this is quite the PR debacle. `</shaking_my_head>`

**Comment by leycec** [comment]
@msarahan: Please reopen this issue, which clearly remains unresolved.

As @jorisvandenbossche [previously noted](https://github.com/conda/conda/issues/8836#issuecomment-507803609), the [supposedly officially supported solution](https://github.com/conda/conda/issues/8836#issuecomment-506388019) of replacing `%PATH%` munging with `call %MINICONDA%\Scripts\activate.bat` fails with a [blatant %PATH% error](https://ci.appveyor.com/project/betse/betse/branch/master/job/sxjny0u1ay840bxl):

```
Build started
ECHO %PYTHON_VERSION% %MINICONDA_DIRNAME%
3.7 C:\Miniconda37-x64
git clone -q --branch=master https://gitlab.com/betse/betse.git C:\projects\betse
git checkout -qf 4936844309e9f9a5569ee79bb488a2001e157d69
Running Install scripts
call %MINICONDA%\Scripts\activate.bat
The system cannot find the path specified.
Command exited with code 1
```

Insert grievously displeased emoji here.

**Comment by leycec** [comment]
@msarahan: Please reopen this issue, which clearly remains unresolved.

As @jorisvandenbossche [previously noted](https://github.com/conda/conda/issues/8836#issuecomment-507803609), the [supposedly officially supported solution](https://github.com/conda/conda/issues/8836#issuecomment-506388019) of replacing `%PATH%` munging with `CALL "%PYTHON_ROOT%\\Scripts\\activate.bat"` ala 44557fc [fails with a blatant %PATH% error](https://ci.appveyor.com/project/betse/betse/branch/master/job/nyu8hsbcuf0lby4s):

```
Build started
ECHO %PYTHON_VERSION% %MINICONDA_DIRNAME%
3.7 C:\Miniconda37-x64
git clone -q --branch=master https://gitlab.com/betse/betse.git C:\projects\betse
git checkout -qf 5f99e59751ba23353167dabc9491c9feb7ef4e26
Running Install scripts
CALL "%PYTHON_ROOT%\\Scripts\\activate.bat"
The network path was not found.
Command exited with code 1
```

**What is the officially supported solution?** Insert grievously displeased emoji here.

**Comment by leycec** [comment]
**Oh, boy.** Due to @msarahan's overly terse responses, I (and presumably @jorisvandenbossche) incorrectly assumed `%PYTHON_ROOT%` to be a universally available environment variable. Of course, it isn't. It's specific to [conda's AppVeyor configuration](https://github.com/conda/conda/blob/master/appveyor.yml).

Of course, that still doesn't help. Incorporating this environment variable into [our AppVeyor configuration](https://gitlab.com/betse/betse/blob/master/.appveyor.yml) then [fails with an absurd error](https://ci.appveyor.com/project/betse/betse/branch/master/job/3c77j5fsmm2b6j0h) in which `conda` can no longer find itself after it itself successfully creates a `conda` environment. <sup>I wish I was kidding.</sup>

Specifically, the following seemingly sensible `.appveyor.yml`:

```
environment:
  matrix:
    - PYTHON_VERSION: 3.7
      PYTHON_ROOT: C:\Miniconda37-x64
    - PYTHON_VERSION: 3.6
      PYTHON_ROOT: C:\Miniconda36-x64
    - PYTHON_VERSION: 3.5
      PYTHON_ROOT: C:\Miniconda35-x64

install:
  - CALL "%PYTHON_ROOT%\\Scripts\\activate.bat"
  - conda config --set always_yes true
  - conda config --add channels conda-forge
  - conda update --quiet conda
  - conda info --all
  - conda create --quiet --name conda-env-%PYTHON_VERSION% python=%PYTHON_VERSION%
  - conda activate conda-env-%PYTHON_VERSION%
  - conda install --quiet --file requirements-conda.txt
  - python setup.py develop
```

...raises [the following fatal error](https://ci.appveyor.com/project/betse/betse/branch/master/job/3c77j5fsmm2b6j0h):

```
...
conda activate conda-env-%PYTHON_VERSION%
conda install --quiet --file requirements-conda.txt
'conda' is not recognized as an internal or external command,
operable program or batch file.
Command exited with code 1
```

Let's summarize what we've learned, shall we? Activating a newly created environment appears to silently succeed, but *actually* destroys the current `%PATH%` in a sufficiently horrible manner that the `conda` command no longer exists.

Please be honest: *was any of this actually tested by a living human being?*

**Comment by leycec** [comment]
In hindsight, we should have just ignored @msarahan and 44557fc. Officially supported solutions that don't actually work are officially useless. Instead, we've adopted the same approach taken by *literally everyone else in the world.* We've reverted to [our prior AppVeyor configuration](https://gitlab.com/betse/betse/blob/master/.appveyor.yml) with an additional `activate` call added before the call to `conda info --all` and replaced all calls to `conda activate` with merely `activate`.

Specifically, **just do this**:

```
environment:
  matrix:
    - PYTHON_VERSION: 3.7
      PYTHON_ROOT: C:\Miniconda37-x64
    - PYTHON_VERSION: 3.6
      PYTHON_ROOT: C:\Miniconda36-x64
    - PYTHON_VERSION: 3.5
      PYTHON_ROOT: C:\Miniconda35-x64

install:
  # Do this. Don't CALL "%PYTHON_ROOT%\\Scripts\\activate.bat".
  - set "PATH=%PYTHON_ROOT%;%PYTHON_ROOT%\\Scripts;%PATH%;"
  - conda config --set always_yes true
  - conda config --add channels conda-forge
  - conda update --quiet conda
  # Do this, too. Yes, this is insane.
  - activate
  - conda info --all
  - conda create --quiet --name conda-env-%PYTHON_VERSION% python=%PYTHON_VERSION%
  # Do this. Don't "conda activate conda-env-%PYTHON_VERSION%".
  - activate conda-env-%PYTHON_VERSION%
  - conda install --quiet --file requirements-conda.txt
  - python setup.py develop
```

Anaconda developers, you have much to answer for. :face_with_head_bandage:

**Comment by vjwada** [comment]
@msarahan said "libarchive DLLs are not getting found. Are you activating the environment with Conda in it?",
But in "manage with conda" dialog in pydev in Eclipse "conda activate" returned same message "TypeError: LoadLibrary() argument 1 must be str, not None".
libarchive was installed and all conda environments were already updated, and all of conda command were succeeded in CLI.

How to activate that conda environment in environment that can not be activated?
Where's the problem? pydev environment?

**Comment by jorisvandenbossche** [comment]
@leycec please stop with your personal attacks towards @msarahan. Yes, breakages like this are annoying, I have been wasting time on it as well, but that is no reason to be so aggressive towards a single person, neither will it help solve the problem. @msarahan has been helping on such issues in the past very constructively, and is putting a lot of hard work in conda, of which we are all benefiting. 

That said, it would indeed be helpful is there was a bit more guidance on those issues of how to use conda in appveyor, or how to solve the breakages.

**Comment by msarahan** [comment]
> Please be honest: was any of this actually tested by a living human being?

Yes, clearly at least one, because our Appveyor configuration uses it.

> Anaconda developers, you have much to answer for. 🤕

Like how much time we've saved you over the years?  Like how much time we spend doing FREE support on top of all the development time that goes into things?  Are you entitled to anything at all here?  That's rhetorical, but the license spells it out if you prefer:

```
THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```

Perhaps you should start from a position of gratefulness, and work backwards to "things aren't working well, is there anything I can do to help?"  Your negativity is not helping anything.  Go build a package manager and compile an ecosystem for yourself.  Come back when you appreciate the magnitude of what you're getting for free.

The best way to activate base environments is to ``call`` the activate.bat script.  This script knows to use the conda.bat file in $PREFIX/condabin.  This .bat file in turn sets up PATH so that the required libraries get found.

``call C:\Miniconda3\Scripts\activate``

After this, the ``conda activate`` commands should work correctly.

The main problem here is that everyone has been assuming that two PATH entries is enough.  conda.exe is a python entry point EXE, and it's not easy or good to customize it with extra content.  That's why we have it in the .bat scripts.

Additionally, each CI provider is different in how state is or is not shared between sections.  You may need to run ``call C:\Miniconda3\Scripts\activate`` at the start of each section that you want to you conda in.

**Comment by jorisvandenbossche** [comment]
> call C:\Miniconda3\Scripts\activate
>
> After this, the conda activate commands should work correctly.

@msarahan thanks for the additional explanation. 
I tried this, but don't yet get it working. Would you be able to take a quick look at https://github.com/geopandas/geopandas/pull/1029 in case you see something obviously wrong?

**Comment by msarahan** [comment]
Yes, I'll take a look at the CI when it finishes.  I didn't see anything wrong - the error was unsatisfiable, not that conda failed to run.  I think you might be misusing the CONDA_RESTORE_FREE_CHANNEL - windows syntax doesn't allow you to inline env var settings, to the best of my knowledge

**Comment by jorisvandenbossche** [comment]
The py2 build on appveyor is indeed failing with unsatisfiable error, but the py3 build not, and there I still get "'conda' is not recognized as an internal or external command,": https://ci.appveyor.com/project/jorisvandenbossche/geopandas-us6w0/builds/25778050/job/h8ky7hgv001a9gqh

**Comment by raphaelreis** [comment]
Hi, 
Just in case. I got the issue when trying to install pkg through the *powershell prompt*. However, everthing goes well with the *normal* prompt from Anaconda

**Comment by HugoBarreto** [comment]
I've just installed Anaconda3-2019.03-Windows-x86_64 and conda was fine, then updated conda, now it's crashing. What have I missed??

```bash
The following packages will be UPDATED:

  conda                                       4.6.11-py37_0 --> 4.7.10-py37_0

```

**Comment by msarahan** [comment]
It's not a problem with 4.6 because 4.7 adds the libarchive dependency.  This makes conda more sensitive to PATH on Windows especially, because PATH is how library lookup happens.

Your fix of `conda init bash` is a good one, because it activates the base environment by default at startup.  Activation is definitely the optimal way to use conda (as opposed to manually changing PATH) because your packages may set environment variables when your environment is activated.  This won't happen with PATH modifications, and those packages often won't work.  GDAL and java are two that come to mind.

If anyone is interested, there are preview anaconda installers at https://repo.continuum.io/pkgs/misc/previews/anaconda/2019.07/ and miniconda at https://repo.continuum.io/pkgs/misc/previews/miniconda/4.7.10/

**Comment by Ayeeta** [comment]
Use Anaconda prompt

> Hey all, ive been trying to do :
> conda create -n myenv python=3.5.1
> or conda list
> 
> '''
> **TypeError: LoadLibrary() argument 1 must be str, not None''**
> '''
> 
> ## Expected Behavior:
> new env creation..
> or
> list of all conda pkgs
> 
> (when i looked for this bugs it gave answers for specific issues it was caused by nothing that i understood could help me)
> 
> [details:
> windows 10
> conda 4.7.5
> Python 3.7.3]
> 
> ## Environment Information
> `` `conda info` ``
> ```
> 
> # >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<
> 
>     Traceback (most recent call last):
>       File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda\exceptions.py", line 1043, in __call__
>         return func(*args, **kwargs)
>       File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda\cli\main.py", line 84, in _main
>         exit_code = do_call(args, p)
>       File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda\cli\conda_argparse.py", line 80, in do_call
>         module = import_module(relative_mod, __name__.rsplit('.', 1)[0])
>       File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\importlib\__init__.py", line 127, in import_module
>         return _bootstrap._gcd_import(name[level:], package, level)
>       File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
>       File "<frozen importlib._bootstrap>", line 983, in _find_and_load
>       File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
>       File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
>       File "<frozen importlib._bootstrap_external>", line 728, in exec_module
>       File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
>       File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda\cli\main_info.py", line 19, in <module>
>         from ..core.index import _supplement_index_with_system
>       File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda\core\index.py", line 9, in <module>
>         from .package_cache_data import PackageCacheData
>       File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda\core\package_cache_data.py", line 15, in <module>
>         from conda_package_handling.api import InvalidArchiveError
>       File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\conda_package_handling\api.py", line 3, in <module>
>         from libarchive.exception import ArchiveError as _LibarchiveArchiveError
>       File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\libarchive\__init__.py", line 1, in <module>
>         from .entry import ArchiveEntry
>       File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\libarchive\entry.py", line 6, in <module>
>         from . import ffi
>       File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\site-packages\libarchive\ffi.py", line 27, in <module>
>         libarchive = ctypes.cdll.LoadLibrary(libarchive_path)
>       File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\ctypes\__init__.py", line 434, in LoadLibrary
>         return self._dlltype(name)
>       File "C:\Users\asaelb\AppData\Local\Continuum\anaconda3\lib\ctypes\__init__.py", line 356, in __init__
>         self._handle = _dlopen(self._name, mode)
>     TypeError: LoadLibrary() argument 1 must be str, not None
> 
> `$ C:\Users\asaelb\AppData\Local\Continuum\anaconda3\Scripts\conda-script.py info`
> 
> 
> An unexpected error has occurred. Conda has prepared the above report.
> 
> Upload successful.
> ```
> 
> `` `conda config --show-sources` ``
> ```
> C:\Users\asaelb>conda config --show-sources
> ==> C:\Users\asaelb\.condarc <==
> ssl_verify: False
> channels:
>   - defaults
>   - conda-forge
> report_errors: True
> ```
> 
> `` `conda list --show-channel-urls` ``

use Anaconda prompt

**Comment by hellmrf** [comment]
When running `conda <command> <package>`, the correct file to execute is `[conda]\condabin\conda.bat`. However, as the installation root have a `conda.exe`, it will be executed instead of the right file, causing the crash. 

So run (assuming that you've installed conda at `C:\Anaconda\`):

```
C:\> cd Anaconda\condabin\
C:\Anaconda\condabin\> conda.bat list
```ing

**Comment by sherif-med** [comment]
This problem can raise when you miss configure your path variable.
Add all of the four listed below to the path variable:
*\Anaconda3;
*\Anaconda3\Scripts;
*\Anaconda3\libs\python37;
*\Anaconda3\Library\bin

**Comment by darlove** [comment]
This has worked for me: https://conda.io/projects/conda/en/latest/user-guide/troubleshooting.html#condahttperror-http-000-connection-failed

Do what they ask you to in the solution section. Basically, you have to run the command that sets ssl_verify to false (but through the command, not editing the file) and use the flag --system. Not sure if this is required but that's what I've done. Once done, I've closed the terminal (anaconda prompt) and reopened. Then ran conda update conda and it worked. I ran the terminal in administrator mode.

Hope this helps others as well.

Best
Darek

**Comment by kuriakinzeng** [comment]
I had the same problem. Adding <path-to-Anaconda>/Library/bin to PATH works for me.

**Comment by duskmoon314** [comment]
> This problem can raise when you miss configure your path variable.
> Add all of the four listed below to the path variable:
> *\Anaconda3;
> *\Anaconda3\Scripts;
> *\Anaconda3\libs\python37;
> *\Anaconda3\Library\bin

I just tried this and it works for me. Well, there is a tiny difference that I just add three of them:

```
*\Anaconda3
*\Anaconda3\Scripts
*\Anaconda3\Library\bin
```

 @sherif-med, thank you.

**Comment by OrsoEric** [comment]
Just had the same problem, I couldn't install packages from pyCharm integrated package manager. A search landed me here, and most answers didn't help.  @hellmrf solution was spot on. PyCharm calls the wrong executable by default on the package manager window. 

Command line called by PyCharm integrated package manager window: (Doesnt work)
C:\ProgramData\Anaconda3\Scripts\conda.exe install -p C:/ProgramData/Anaconda3/envs/TestPlot matplotlib -y

Correct command line: (works)
C:\ProgramData\Anaconda3\condabin\conda.bat install -p C:/ProgramData/Anaconda3/envs/TestPlot matplotlib -y

Problem with this solution is that i'm forced to use the terminal. I'd like to restore the package manager window. is there any way to have it call the right command?

**Comment by msarahan** [comment]
Modifying PATH may be the best band-aid to get PyCharm or other IDE's that aren't coded to use activation working. VS Code does do activation properly, and newer versions of PyCharm do also, I think.

**Comment by harshkaso** [comment]
> Yep, that is the problem. Scripts has conda.exe, not conda.bat, but conda.bat is the thing that sets up the correct library paths.

Exactly, I simply added the path of condabin to environment variables and it worked for me.

**Comment by mingwandroid** [comment]
Adding just condabin is ok. Adding anything else is strongly discouraged.

**Comment by pschaeffer** [comment]
I had the same problem. I was able to get around it by installing the newest release of Anaconda3 and by using the Anaconda prompt. The Anaconda prompt has the base environment set (at least for me) and has (already has) the PATH variable updated. This combination worked (at least for me). Note that I am running Windows 10 and used the conda command, not _conda.

**Comment by MotiBaadror** [comment]
I had the same issue. 
Machine :Windows 
In my case adding below path to the environment variable worked.
```
anaconda/Library/bin
```

**Comment by Devanshu-Soni-0** [comment]
In my case below path worked:
C:\Users\Asus\Anaconda3\Library\bin
